tests/pytests/test_dialect.py

```diff
@@ -43,36 +43,179 @@ def test_dialect_query_errors(env):
+    env.expect('FT.EXPLAIN', 'idx', '@num:[-inf  inf]', 'DIALECT', 5).contains('NUMERIC {-inf <= @num <= inf}\n')
+
+    env.expect('FT.EXPLAIN', 'idx', "@t1:(w'abc?')", 'DIALECT', 1).contains('INTERSECT')
+    env.expect('FT.EXPLAIN', 'idx', "@t1:(w'abc?')", 'DIALECT', 2).contains('@t1:WILDCARD{abc?}')
+    env.expect('FT.EXPLAIN', 'idx', "@t1:(w'abc?')", 'DIALECT', 5).contains('@t1:WILDCARD{abc?}')
+
+    env.expect('FT.EXPLAIN', 'idx', "@t1:(*a\\w's')", 'DIALECT', 1).contains('@t1:INTERSECT {\n  @t1:SUFFIX')
+    env.expect('FT.EXPLAIN', 'idx', "@t1:(*a\\w's')", 'DIALECT', 2).contains('@t1:INTERSECT {\n  @t1:SUFFIX')
+    env.expect('FT.EXPLAIN', 'idx', "@t1:(*a\\w's')", 'DIALECT', 5).error().contains('Syntax error')
+
+    env.expect('FT.EXPLAIN', 'idx', "@t1:(w'*')", 'DIALECT', 1).error().contains('Syntax error')
+    env.expect('FT.EXPLAIN', 'idx', "@t1:(w'*')", 'DIALECT', 2).contains('@t1:WILDCARD{*}')
+    env.expect('FT.EXPLAIN', 'idx', "@t1:(w'*')", 'DIALECT', 5).contains('@t1:WILDCARD{*}')
+
+    env.expect('FT.EXPLAIN', 'idx', "*-*", 'DIALECT', 1).error().contains('Syntax error')
+    env.expect('FT.EXPLAIN', 'idx', "*-*", 'DIALECT', 2).error().contains('Syntax error')
+    env.expect('FT.EXPLAIN', 'idx', "*-*", 'DIALECT', 5).error().contains('Syntax error')
+
+    env.expect('FT.EXPLAIN', 'idx', "*1*", 'DIALECT', 1).contains('INFIX{*1*}')
+    env.expect('FT.EXPLAIN', 'idx', "*1*", 'DIALECT', 2).contains('INFIX{*1*}')
+    env.expect('FT.EXPLAIN', 'idx', "*1*", 'DIALECT', 5).contains('INFIX{*1*}')
+
+    env.expect('FT.EXPLAIN', 'idx', "abc!", 'DIALECT', 1).contains('UNION {\n  abc\n  +abc(expanded)\n}\n')
+    env.expect('FT.EXPLAIN', 'idx', "abc!", 'DIALECT', 2).contains('UNION {\n  abc\n  +abc(expanded)\n}\n')
+    env.expect('FT.EXPLAIN', 'idx', "abc!", 'DIALECT', 5).contains('Syntax error')
+
+    res = env.cmd('FT.EXPLAINCLI', 'idx', "1.2e+3", 'DIALECT', 1)
+    expected = [
+      'INTERSECT {',                                                                                                                                 
+      '  UNION {',
+      '    1.2',
+      '    +1.2(expanded)',
+      '  }',
+      '  UNION {',
+      '    e',
+      '    +e(expanded)',
+      '  }',
+      '  UNION {',
+      '    3',
+      '    +3(expanded)',
+      '  }',
+      '}',
+      ''
+    ]
+    env.assertEqual(res, expected)
+    res = env.cmd('FT.EXPLAINCLI', 'idx', "1.2e+3", 'DIALECT', 2)
+    expected = [
+      'INTERSECT {',
+      '  UNION {',
+      '    1.2',
+      '    +1.2(expanded)',
+      '  }',
+      '  INTERSECT {',
+      '    UNION {',
+      '      e',
+      '      +e(expanded)',
+      '    }',
+      '    UNION {',
+      '      3',
+      '      +3(expanded)',
+      '    }',
+      '  }',
+      '}',
+      ''
+    ]
+    env.assertEqual(res, expected)
+    res = env.cmd('FT.EXPLAINCLI', 'idx', "1.2e+3", 'DIALECT', 5)
```

Comment:
Can we avoid expansion/stemming from numeric values?
(`opts.flags |= QueryNode_Verbatim`)

Reply:
I create a ticket for it: MOD-7025
The change should consider parameter evaluation also:
```sh
127.0.0.1:6379> FT.EXPLAINCLI idx $p dialect 2 PARAMS 2 p 2.3
1) UNION {
2)   2.3
3)   +2.3(expanded)
4) }
```